routers/repo/http.go

```diff
@@ -156,9 +156,11 @@ func HTTP(ctx *context.Context) {
 					ctx.Handle(http.StatusInternalServerError, "UserSignIn error: %v", err)
 					return
 				}
+			}
 
-				// Assume username now is a token.
-				token, err := models.GetAccessTokenBySHA(authUsername)
+			if authUser == nil {
+				// Assume password is a token.
+				token, err := models.GetAccessTokenBySHA(authPasswd)
```

Comment:
You need to also check if user `models.GetUserByName(authUsername)` returns correct user and validate that returned user.ID is equal to token.UID